common/parse.go

```diff
@@ -73,3 +75,100 @@ func ExtractLowercaseFieldsFromRaw(fields []string, record map[string]interface{
 
 	return out
 }
+
+var (
+	ErrNotArray   = errors.New("results is not an array")
+	ErrNotObject  = errors.New("result is not an object")
+	ErrNotString  = errors.New("link is not a string")
+	ErrNotNumeric = errors.New("value is not numeric")
+	ErrNotInteger = errors.New("value is not integer")
+
+	// JSONManager is a helpful wrapper of ajson library that adds errors when querying JSON payload
+	// and provides common conversion methods.
+	JSONManager = jsonManager{}
```

Comment:
Querying JSON content deals with finding "keys" and validating expected data type of the "value".
ErrNotArray, ErrNotObject errors can be universally shared for all JSON parsers